frontend/packages/modelServing/modelRegistry/useDeployButtonState.ts

```diff
@@ -0,0 +1,22 @@
+import { SupportedArea } from '@odh-dashboard/internal/concepts/areas/types';
+import useIsAreaAvailable from '@odh-dashboard/internal/concepts/areas/useIsAreaAvailable';
+import useServingPlatformStatuses from '@odh-dashboard/internal/pages/modelServing/useServingPlatformStatuses';
+import { getDeployButtonState } from '@odh-dashboard/internal/pages/modelCatalog/utils';
+
+/**
+ * Returns the deploy button state (visible, enabled, tooltip) for model deploy actions.
+ */
+const useDeployButtonState = (): { visible: boolean; enabled?: boolean; tooltip?: string } => {
+  const isModelServingEnabled = useIsAreaAvailable(SupportedArea.MODEL_SERVING).status;
+  const { platformEnabledCount, kServe } = useServingPlatformStatuses();
```

Comment:
There is a model serving plugin version of this function, it should be fully equivalent to `useServingPlatformStatuses`. It is slightly wordier
```suggestion
  const { clusterPlatforms } = useAvailableClusterPlatforms();  
  const platformEnabledCount = clusterPlatforms.length();
  const kServe === clusterPlatforms.find((p) => p.properties.id === KSERVE_ID); // KSERVE_ID is "kserve"
```